app/controllers/foreman_tasks/tasks_controller.rb

```diff
@@ -72,17 +72,15 @@ def resume
     def unlock
       task = find_dynflow_task
       if task.paused?
-        task.state = :stopped
-        task.save!
-        render json: { statusText: 'OK' }
+        force_unlock(task)
       else
         render json: {}, status: :bad_request
       end
     end
 
-    def force_unlock
-      task       = find_dynflow_task
+    def force_unlock(task = find_dynflow_task)
```

Comment:
ugh... controller action with a param smells bit wrong to me. Would it be possible to use `@task` in unlock instead?
then the first line here would be `@task ||= find_dynflow_task` which smells bit better to me.

Or possibly just extract the unlocking into a separate method instead of calling one controller action from the other? it would be the cleanest way IMHO

Reply:
Extracted to a separate method + replaced all the calls to find_dynflow_task with a before_action